#Polishing Patch - Clown Rat

ItsDzordz opened the pull request:
# Description

This pr aims to polish up the clown rat.
Changes:
- Made the nightvision on base mouse togglable for better accesbility.
- Added sperate Ghost role categories for the clown rat and crazed clown rat, making fiding them in ghost roles easier
- Fixed the butchering bug on crazed rats, when they would upon butchering turn to regular clown rats.
- Adjusted the spliting thresholds for 1-st generation crazed clown rats (lowered psychal tresholds by 10 to guarntee the first split due to mixed damage, while rasing the burn threshold by 5. the first generation dying without spliting is a huge let down for the price tag, so this aims to make it more consistent)
- Changed the sound of the clown rite for ma slam to a bite (its more accurate and less annoying to hear when there is a lot of rats)
- Added 4 bleed damage and extra 1 pierce damage to crazed clown rats, to justify the price tag of 10tc

Before patch damage:
        Slash: 2
        Blunt: 2
        Piercing: 2

After patch damage:
        Slash: 2
        Blunt: 2
        Piercing: 3
        Bloodloss: 4

---

# TODO

- [x] Polish up the clown rats

---

# Changelog


:cl:

- tweak: Tweaked crazed clown rats

ItsDzordz committed:
Clown rat polishing patch

Mnemotechnician commented on Resources/Prototypes/_Floof/Entities/Mobs/NPCs/clown_abominations.yml:
```diff
@@ -346,6 +393,22 @@
+    name: Crazed Clown Rat
+    description: A plague released upon the world by the Syndicate, go be a menace or bite someones ankle.
+    rules: floof-ghost-role-information-team-antag-rules 
       
```
Notice how this block is repeated 3 times in this file. You could create an abstract parent entity and define those components in it, thus avoiding having to duplicate code. But it's not a significant issue.

ItsDzordz replied:
Can i crate a parent entity that is jsut these 2 types, and it would not appier as a sperate entity in the spawn menu?

Mnemotechnician replied:
`abstract: true` would prevent the entity from being spawned by *any* means, appearing in the spawn menu, or being checked by tests (e.g. you can have an abstract entity without a valid sprite)

ItsDzordz replied:
Applied optimazation

Mnemotechnician reviewed (approved):
Personally I think bloodloss damage is a weird choice because clown rats don't like,... suck your blood out, do they? However it's not an issue either.

ItsDzordz commented:
> Personally I think bloodloss damage is a weird choice because clown rats don't like,... suck your blood out, do they? However it's not an issue either.

Carps dotn suck blood eather, but they do bloodloss due to jsut how sharp their teeth are, i think rats would behave in similar way, more holes and bloodloss than actuall up front damge, and givign them bloodloss, make them a better tool for over all cripling and chaos, since even if a fight is won by the victims bloodloss will slow down the recovery of the victims, buying more time as a distraction.

ItsDzordz committed:
Optimazation patch

~ Optimized the clown rat code.
+ made the nightvision on all mouse toggleable (for better accesbility)

ItsDzordz commented:
All optimization complete, test complete, its good to go
